nfpm.go

```diff
@@ -30,12 +36,66 @@ func Get(format string) (Packager, error) {
 	return p, nil
 }
 
+// Parse decodes YAML data from an io.Reader into a configuration struct
+func Parse(in io.Reader) (config Config, err error) {
+	dec := yaml.NewDecoder(in)
+	dec.SetStrict(true)
+	err = dec.Decode(&config)
+	return
+}
+
+// ParseFile decodes YAML data from a file path into a configuration struct
+func ParseFile(path string) (config Config, err error) {
+	var file *os.File
+	file, err = os.Open(path)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+	return Parse(file)
+}
+
 // Packager represents any packager implementation
 type Packager interface {
 	Package(info Info, w io.Writer) error
 }
 
-// Info contains information about the package
+// Config contains the top level configuration for packages
+type Config struct {
+	Info      `yaml:",inline"`
+	Overrides map[string]interface{} `yaml:"overrides,omitempty"`
```

Comment:
> may contain additional packager exclusive configurations 

I think that since this is not a thing right now, we can leave as `Info`.

When we start to need custom fields per-format, than we can always change this to `interface{}` anyway... 

what o you think?

Reply:
My plan was to enable `rpm_pretrans` and `rpm_posttrans` as custom fields. Either we have to lose them (go with `Info`) or use `intreface{}`.

Reply:
or maybe a `map[string]string`?

Reply:
yes but then overriding field like *Depends* which is `[]string` is not possible right?

Reply:
I mean, do something like my comment below for overrides and this `map[string]string` for package specifics...

Is it safe to assume that packager-specific fields will always be a string?

(thinking that most features exists in both rpms and debs)

Reply:
Not a bad idea actually. This is probably the way to go. I will commit an implementation of this and we can discuss from there 👍